include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -365,6 +377,8 @@ inline void NSGA2::FastNonDominatedSort(
     i++;
     fronts.push_back(nextFront);
   }
+  //Remove the empty final set
+  fronts.pop_back();
```

Comment:
Looks like this causes a bad alloc.

Reply:
The bad alloc is because I used ```!fronts.empty()``` instead of ```!fronts[i].empty()```. Lemme fix that ;)